src/network/clientpackethandler.cpp

```diff
@@ -1233,28 +1233,129 @@ void Client::handleCommand_HudSetParam(NetworkPacket* pkt)
 
 void Client::handleCommand_HudSetSky(NetworkPacket* pkt)
 {
-	std::string datastring(pkt->getString(0), pkt->getSize());
-	std::istringstream is(datastring, std::ios_base::binary);
+	video::SColor bgcolor;
+	std::string type;
+	bool clouds;
+	u16 texture_count;
+	std::vector<std::string> params;
+	std::string texture;
+	
+	video::SColor day_sky;
+	video::SColor day_horizon;
+	video::SColor dawn_sky;
+	video::SColor dawn_horizon;
+	video::SColor night_sky;
+	video::SColor night_horizon;
+	video::SColor indoors;
+	video::SColor sun_tint;
+	video::SColor moon_tint;
+	std::string tint_type;
+
+	*pkt >> bgcolor >> type >> clouds;
+
+	if (type == "skybox") {
+		*pkt >> texture_count;
+
+		for (int i = 0; i < texture_count; i++) {
+			*pkt >> texture;
+			params.emplace_back(texture);
+		}
+	} else if (type == "regular") {
+		*pkt >> day_sky >> day_horizon
+			>> dawn_sky >> dawn_horizon
+			>> night_sky >> night_horizon
+			>> indoors;
+	}
```

Comment:
for easier forwards-compatibility the client should abort (or do something other useful) when encountering an unknown type here